src/factory-daemon.js

```diff
@@ -0,0 +1,119 @@
+'use strict'
+
+const defaults = require('lodash.defaultsdeep')
+const clone = require('lodash.clone')
+const waterfall = require('async/waterfall')
+const path = require('path')
+
+const Daemon = require('./ipfsd-daemon')
+const defaultConfig = require('./defaults/config')
+const defaultOptions = require('./defaults/options')
+
+/**
+ * Spawn IPFS Daemons (either JS or Go)
+ *
+ * @namespace FactoryDaemon
+ */
+class FactoryDaemon {
+  /**
+   *
+   * @param {Object} opts
+   *  - `type` string - 'go' or 'js'
+   *  - `exec` string (optional) - the path of the daemon executable
+   * @return {*}
+   */
+  constructor (options) {
+    if (options && options.type === 'proc') {
+      throw new Error('This Factory does not know how to spawn in proc nodes')
+    }
+    options = Object.assign({ type: 'go' }, options)
+    this.type = options.type
+    this.exec = options.exec
+  }
+
+  /**
+   * Get the version of the currently used go-ipfs binary.
+   *
+   * @memberof FactoryDaemon
+   * @param {Object} [options={}]
+   * @param {function(Error, string)} callback
+   * @returns {undefined}
+   */
+  version (options, callback) {
+    if (typeof options === 'function') {
+      callback = options
+      options = {}
+    }
+    options = Object.assign({}, options, { type: this.type })
+    // TODO: (1) this should check to see if it is looking for Go or JS
+    // TODO: (2) This spawns a whole daemon just to get his version? There is
```

Comment:
- [ ] 🌟

Reply:
just doing `new Daemon` won't actually start the daemon, it will simply allow you to send commands to the selected executable.

Reply:
It still should check if it is JS or Go

Reply:
Why? All the logic around daemon type is already being handled in the `Daemon` itself - not sure what to do with the check there.